src/Symfony/Bundle/SecurityBundle/DependencyInjection/MainConfiguration.php

```diff
@@ -399,7 +399,7 @@ private function addEncodersSection(ArrayNodeDefinition $rootNode)
                         ->performNoDeepMerging()
                         ->beforeNormalization()->ifString()->then(function ($v) { return ['algorithm' => $v]; })->end()
                         ->children()
-                            ->scalarNode('algorithm')->cannotBeEmpty()->end()
+                            ->scalarNode('algorithm')->cannotBeEmpty()->validate()->ifTrue()->thenInvalid('Invalid algorithm %s')->end()->end()
```

Comment:
what about validating that the value is a string instead?

Reply:
I can do it like this:

```
->scalarNode('algorithm')
	->cannotBeEmpty()
	->validate()
		->ifTrue(function ($v) { return !\is_string($v); })
		->thenInvalid('Invalid algorithm %s')
	->end()
->end()
```

But it is not very elegant. Is there a beter way of validating this?

Reply:
Good suggestion!